### PR TITLE
Added ArtStation support

### PR DESCRIPTION
**Pull request by dearrrfish**
e.g https://www.artstation.com/artwork/Aq3Ygm

### CODE REVIEW

File: nazurin/sites/Artstation/api.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from nazurin.utils import Request
+from nazurin.utils.exceptions import NazurinError
+
+class Artstation(object):
+    async def getPost(self, post_id: int):
+        """Fetch an post."""
+        api = f"https://www.artstation.com/projects/{post_id}.json"
+        async with Request() as request:
+            async with request.get(api) as response:
+                if not response.text:
```
Comment (y-young) [inline nazurin/sites/Artstation/api.py:14]: It would be better by checking if the response code is 404.

File: nazurin/sites/Artstation/config.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+PRIORITY = 11
```
Comment (y-young) [inline nazurin/sites/Artstation/config.py:1]: Let's set the priority the same as Pixiv, which is 10.

File: nazurin/sites/Artstation/api.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+            return:
+                '_z-ed_-da.jpg',
+                'https://cdnb.artstation.com/p/assets/images/images/042/908/363/large/_z-ed_-da.jpg',
+                'https://cdnb.artstation.com/p/assets/images/images/042/908/363/small_square/_z-ed_-da.jpg'
+
+        """
+        baseurl = src.split('?')[0]
+        basename = os.path.basename(baseurl)
+        thumbnail = baseurl.replace('/large/', '/small_square/')
```
Comment (y-young) [inline nazurin/sites/Artstation/api.py:73]: Normally we want to keep the heigh-width ratio, so "medium" is a better option, e.g: https://cdnb.artstation.com/p/assets/covers/images/042/647/153/medium/_z-ed_-z-ed-da2.jpg

File: nazurin/sites/Artstation/api.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    def parseUrl(self, src: str) -> Tuple[str, str, str]:
+        """Get filename, original file url & thumbnail url of the original image
+
+        eg:
+
+            src: 'https://cdnb.artstation.com/p/assets/images/images/042/908/363/large/_z-ed_-da.jpg?1635784439'
+
+            return:
+                '_z-ed_-da.jpg',
+                'https://cdnb.artstation.com/p/assets/images/images/042/908/363/large/_z-ed_-da.jpg',
```
Comment (y-young) [inline nazurin/sites/Artstation/api.py:67]: I noticed that "large" is not necessarily the image of the best quality we can get, and the size of "large" image doesn't match that in "assets", "4k" seems a better option, e.g: https://cdna.artstation.com/p/assets/images/images/042/647/078/4k/_z-ed_-da1.jpg?1635092391

**Review by y-young — CHANGES_REQUESTED** [review]
Thanks for your contribution! The code looks good overall, but it can be better with some adjustments, please see the comments.

File: nazurin/sites/Artstation/api.py (hunk 1 above)
Comment (dearrrfish) [reply nazurin/sites/Artstation/api.py]: sure, fixed.

File: nazurin/sites/Artstation/api.py (hunk 4 above)
Comment (dearrrfish) [reply nazurin/sites/Artstation/api.py]: Good to know! Thank you!

**Review by y-young — APPROVED** [review]

**Comment by y-young** [comment]
Thank you!

**Pull request merged**